### DISCUSSION TITLE
Window Buttons in Panel

### DISCUSSION THREAD

**Discussion by toboggan7** [lxqt/lxqt#2237] [Q&A] [answered]
I've been thinking of switching to LXQt, but I want to be able to have my close, minimize and maximize buttons in a panel, similar to the Unity desktop environment. I've been able to do this on Plasma via the [Window Buttons](https://store.kde.org/p/1272871/) applet, but all I could find for LXQt was a [Lubuntu feature request from 2014](https://askubuntu.com/questions/439100/lxpanel-show-active-window-title-close-min-max-restore-buttons) (when it still used LXDE).

Is there any way to do this, or do I need to wait for it to be implemented?

**Comment by tsujan** [comment, marked as answer]
> Is there any way to do this

Since LXQt is WM-agnostic (= any window manager can be used under LXQt), no.

I've made a simple plugin for my personal use but it's only for KWin and very special circumstances. So, I didn't and won't make a PR for it. It's useless with other WMs — it's even useless with KWin in general.

![buttons](https://user-images.githubusercontent.com/981076/167676137-3a253114-b951-463c-8d3f-de480a8183f3.png)

To support a Wayland session, LXQt might need to choose a WM and stop being WM-agnostic. Then this feature might be possible. But that won't happen in near future.

**Reply by toboggan7** [reply]
That's a shame. I was specifically meaning with the default choice of Openbox. I know you've said it's not possible, but, targeting Openbox specifically, would it be possible to maybe make a quick launch bar with .desktops running commands that minimize, maximize or close the current window?

**Reply by tsujan** [reply]
> ... the default choice of Openbox

There's no default choice anymore.

**Reply by toboggan7** [reply]
I didn't know that. Just assumed there was since Openbox is in the lxqt package group on Arch and is the default on Lubuntu.

**Reply by tsujan** [reply]
The problem isn't about adding buttons to the panel for closing, minimizing,... the active (maximized) window — that can be done easily — but about telling the WM to remove its buttons or title-bar for that window simultaneously. When the latter can't be done in a general way, the former is useless.

Task buttons can already do more.

**Reply by selairi** [reply]
@ytfcon You can closing, minimizing,... clicking, double clicking or right click over task buttons. But If you want the close, minimize and maximize buttons on taskbar:

I usually remove the title bar of windows, Openbox lets do it with right click over title bar. This trick lets you get more space on screen to use:

![screen](https://user-images.githubusercontent.com/1163139/167724200-e2f8bb70-a903-40d3-b4d9-d37fc5386312.png)

In the screenshot you can see that I have the close, minimize and maximize buttons on taskbar. You need install xdotool and openbox and download these files: 
[maximize.txt](https://github.com/lxqt/lxqt/files/8665099/maximize.txt)
[minimize.txt](https://github.com/lxqt/lxqt/files/8665110/minimize.txt)
[close.txt](https://github.com/lxqt/lxqt/files/8665111/close.txt)

Change the .txt extension to .desktop (github doesn't let me upload desktop files). You will have these files:
![screen2](https://user-images.githubusercontent.com/1163139/167725251-086eebe2-6244-406c-ae03-d88e125de784.png)

Now add Quick Launch widget to your taskbar and drag and drop the 3 desktop files over Quick Lauch widget.

Finally... enjoy it.

PS: If you don't like the icons, edit the desktop files.

**Reply by selairi** [reply]
On wayland, instead xdotool, wtype can be used:

[https://linuxcommandlibrary.com/man/wtype](https://linuxcommandlibrary.com/man/wtype)

**Reply by toboggan7** [reply]
That's exactly what I was looking to do. Thank you.

**Reply by stefonarch** [reply]
In ~/.config/kwinrc (didn't find it in the settings)

```
[Windows]
BorderlessMaximizedWindows=true
...
```

https://user-images.githubusercontent.com/10681413/168164525-ae634810-da75-4927-ac4f-a6febf35ad32.mp4

**Reply by tsujan** [reply]
Yes, that's what I use.

IMO, title-bar buttons on the panel are useful only if maximized (active) windows lose their decorations, while other windows have decorations. That's what KWin and compiz-reloaded can do, in different ways.

**Comment by tsujan** [comment]
https://user-images.githubusercontent.com/981076/168169251-971dacca-e572-400b-a5d1-df7d7546236a.mp4

**Comment by stefonarch** [comment]
It was more for fun - actually I use ctrl+q to close windows, cltr+d  for showing the desktop and hardly change any size of windows, just switch among them, darken the inactive ones.

**Reply by tsujan** [reply]
> It was more for fun

For me too. Task buttons can already do all the needed jobs and more.
